Source/inv.h

```diff
@@ -78,54 +78,81 @@ extern bool invflag;
 extern bool drawsbarflag;
 extern const InvXY InvRect[73];
 
-void FreeInvGFX();
-void InitInv();
+class Inventory {

```

Comment:
This class should be `final`.

Reply:
And since it is only a container for functions its constructor should probably be deleted as well.

Reply:
@Jmgr I have plans to add multiple implementations of this class. Do you think it should be final until then?

Reply:
@Jmgr sorry, I didn't get what constructor are you talking about. `Init()`?

Reply:
Could move the player inventory members to this class and attach it to the player. That would also help to clean up the player structure and possibly allow us to simplify some of the functions here.

Reply:
@AJenbo can we merge this and I  do that in another PR? I am not sure when I will have the time to do so.

Reply:
> @Jmgr I have plans to add multiple implementations of this class. Do you think it should be final until then?

So this class will be inherited? Then no, don't mark it as final, but do set the [destructor as virtual](https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors).

Reply:
> @Jmgr sorry, I didn't get what constructor are you talking about. `Init()`?

I meant that if the class only contains static functions and no state then constructing it makes no sense. But seeing that you intent on using inheritance that it's fine. But why use an `Init()` function instead of a constructor?

Reply:
The method was originally `InitInv()` I just renamed it. I didn't convert to a constructor because the `*inventory` is being initialized still in the Main Menu and the `Init()` is only called just before the in-game screen (so it load the resources).